public/app/features/search/hooks/useSearchQuery.ts

```diff
@@ -12,7 +14,8 @@ import {
 } from '../reducers/actionTypes';
 import { DashboardQuery, SearchLayout } from '../types';
 import { hasFilters, parseRouteParams } from '../utils';
-import { locationService } from '@grafana/runtime';
+
+const updateLocation = debounce((query) => locationService.partial(query), 300);
```

Comment:
Any particular reason for debouncing? Can't we just use `onBlur`?

Reply:
I think debounce might be better, onBlur will require separating state from url (right no url -> search query state -> do search)

Reply:
oki doki, makes sense! Thanks @Clarity-89